test/mini_s3_tests.erl

```diff
@@ -19,112 +19,219 @@
+             {{3600, 900}, {{2015,1,27},{0,45,0}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + 2700))), (?HOUR + 900)}},
+             {{3600, 900}, {{2015,1,27},{0,59,59}}, {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + 2700))), (?HOUR + 900)}},
+             {{3600, 900}, {{2015,1,27},{1,0,0}}  , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + 3600))), (?HOUR + 900)}},
+             {{3600, 900}, {{2015,1,27},{23,44,0}}, {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + ?DAY - 1800))), (?HOUR + 900)}},
+             {{3600, 900}, {{2015,1,27},{23,58,0}}, {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + ?DAY - 900))), (?HOUR + 900)}},
+             {{3600, 900}, {{2015,1,28},{0,0,0}}  , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + ?DAY))), (?HOUR + 900)}},
 
              %% There are 86400 seconds in a day. What happens if the interval is not evenly
              %% divisible in that time? Take 7m for example. 420 secs goes into a day 205.71
              %% times which is a remainder of 300 seconds. We should make sure that we
              %% restart the intervals at midnight, so we don't have day to day drift
 
-             {{3600, 420}, {{2015,1,27},{23,59,0}} , (?MIDNIGHT + ?DAY + ?HOUR)},
-             {{3600, 420}, {{2015,1,28},{0,0,0}}   , (?MIDNIGHT + ?DAY + ?HOUR + 420)},
+             {{3600, 420}, {{2015,1,27},{23,59,0}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + ?DAY - 300))), (?HOUR + 300)}},
+             {{3600, 420}, {{2015,1,28},{0,0,0}}   , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + ?DAY))), (?HOUR + 420)}},
+
+             {{604800, 420}, {{2015,1,27},{0,0,0}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT))), (?WEEK)}}
 
              %% Let's test the old functionality too
-             {3600, {{2015,1,27},{0,0,0}} , (?MIDNIGHT + ?HOUR)},
-             {3600, {{2015,1,27},{0,0,1}} , (?MIDNIGHT + ?HOUR + 1)},
-             {3600, {{2015,1,27},{0,1,1}} , (?MIDNIGHT + ?HOUR + 61)},
-             {3600, {{2015,1,28},{0,1,1}} , (?MIDNIGHT + ?DAY + ?HOUR + 61)}
+%             {3600, {{2015,1,27},{0,0,0}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT))), (?HOUR)}},
+%             {3600, {{2015,1,27},{0,0,1}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT))), (?HOUR + 1)}},
+%             {3600, {{2015,1,27},{0,1,1}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT))), (?HOUR + 61)}},
+%             {3600, {{2015,1,28},{0,1,1}} , {(erlcloud_aws:iso_8601_basic_time(calendar:gregorian_seconds_to_datetime(?MIDNIGHT + ?DAY))), (?DAY + ?HOUR + 61)}}
```

Comment:
Remove?

Reply:
The only reason I kept it is because it's there in `upstream` and thought it would make things easier for us in the future when pulling in changes from `upstream` and/or contributing our changes there.

This said, if we limit the change to a specific commit, it should be safe :)

Reply:
It makes sense to keep it then. You can ignore my comment